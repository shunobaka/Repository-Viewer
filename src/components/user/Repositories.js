/**
 * @fileoverview Defines a Repositories react component that renders a list of
 *    repositories.
 */
import React from 'react';
import { Container, Row, Col, Accordion } from 'react-bootstrap';
import PropTypes from 'prop-types';
import RepositoryItem from './RepositoryItem';

/**
 * Repositories react component that renders a list of repositories.
 * @param {object} props Contains the array of repositories to be displayed
 */
const Repositories = ({ repositories }) => {
  return (
    <Container>
      <Row>
        <Col lg={{ span: 6, offset: 3 }}>
          {repositories.length > 0 ? (
            <Accordion>
              {repositories.map((repo, index) => {
                return (
                  <RepositoryItem
                    key={repo.id}
                    index={index}
                    repository={repo}
                  />
                );
              })}
            </Accordion>
          ) : (
            <div className="alert alert-warning">
              No repositories match the filter!
            </div>
          )}
        </Col>
      </Row>
    </Container>
  );
};

Repositories.propTypes = {
  repositories: PropTypes.array.isRequired,
};

export default Repositories;
