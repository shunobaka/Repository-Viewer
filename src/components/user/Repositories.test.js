/**
 * @fileoverview Defines snapshot unit tests for Repositories component.
 */
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Repositories from './Repositories';

describe('repositories snapshots', () => {
  const mockRepos = [
    {
      name: 'Test',
      id: 0,
      description: 'Description',
    },
    {
      name: 'Test2',
      id: 1,
    },
  ];

  it('mutliple repositories renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Repositories repositories={mockRepos} />
        </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('single repository renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Repositories repositories={[mockRepos[0]]} />
        </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('no repositories renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Repositories repositories={[]} />
        </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
