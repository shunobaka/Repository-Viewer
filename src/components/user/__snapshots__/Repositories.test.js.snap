// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`repositories snapshots mutliple repositories renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-lg-6 offset-lg-3"
    >
      <div
        className="accordion"
      >
        <div
          className="card"
        >
          <div
            className="card-header"
            onClick={[Function]}
          >
            1.  
             
            Test
          </div>
          <div
            aria-expanded={null}
            className="collapse"
          >
            <div
              className="card-body"
            >
              <a>
                Go to repository
              </a>
              <br />
              <br />
              <h5>
                Description:
              </h5>
              <span>
                Description
              </span>
              <br />
              <br />
              <h5>
                Created on:
              </h5>
              <span>
                Invalid Date
              </span>
            </div>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card-header"
            onClick={[Function]}
          >
            2.  
             
            Test2
          </div>
          <div
            aria-expanded={null}
            className="collapse"
          >
            <div
              className="card-body"
            >
              <a>
                Go to repository
              </a>
              <br />
              <br />
              <h5>
                Description:
              </h5>
              <span
                className="text-warning"
              >
                This repository does not have a description
              </span>
              <br />
              <br />
              <h5>
                Created on:
              </h5>
              <span>
                Invalid Date
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`repositories snapshots no repositories renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-lg-6 offset-lg-3"
    >
      <div
        className="alert alert-warning"
      >
        No repositories match the filter!
      </div>
    </div>
  </div>
</div>
`;

exports[`repositories snapshots single repository renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-lg-6 offset-lg-3"
    >
      <div
        className="accordion"
      >
        <div
          className="card"
        >
          <div
            className="card-header"
            onClick={[Function]}
          >
            1.  
             
            Test
          </div>
          <div
            aria-expanded={null}
            className="collapse"
          >
            <div
              className="card-body"
            >
              <a>
                Go to repository
              </a>
              <br />
              <br />
              <h5>
                Description:
              </h5>
              <span>
                Description
              </span>
              <br />
              <br />
              <h5>
                Created on:
              </h5>
              <span>
                Invalid Date
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
